src/utility/svd.jl

```diff
@@ -59,16 +60,24 @@ function TensorKit._compute_svddata!(
     for (c, b) in blocks(t)
         x₀ = randn(eltype(b), size(b, 1))
         howmany = trunc isa NoTruncation ? minimum(size(b)) : blockdim(trunc.space, c)
-        S, lvecs, rvecs, info = KrylovKit.svdsolve(b, x₀, howmany, :LR, alg.alg)
-        if info.converged < howmany  # Fall back to dense SVD if not properly converged
+
+        if howmany / minimum(size(b)) > alg.fallback_threshold  # Use dense SVD for small blocks
             U, S, V = TensorKit.MatrixAlgebra.svd!(b, TensorKit.SVD())
-            Udata[c] = U
-            Vdata[c] = V
-        else  # Slice in case more values were converged than requested
-            Udata[c] = stack(view(lvecs, 1:howmany))
-            Vdata[c] = stack(conj, view(rvecs, 1:howmany); dims=1)
-            S = @view S[1:howmany]
+            Udata[c] = @view U[:, 1:howmany]
+            Vdata[c] = @view V[1:howmany, :]
+        else
+            S, lvecs, rvecs, info = KrylovKit.svdsolve(b, x₀, howmany, :LR, alg.alg)
+            if info.converged < howmany  # Fall back to dense SVD if not properly converged
+                U, S, V = TensorKit.MatrixAlgebra.svd!(b, TensorKit.SVD())
+                Udata[c] = @view U[:, 1:howmany]
+                Vdata[c] = @view V[1:howmany, :]
```

Comment:
Here, the view actually does not make a lot of sense, because the `setindex!` implicitly converts it back to a dense array :wink:

Reply:
Ha, you're right, I need to work on my Julia allocation skills :P